tests/sqlparser_postgres.rs

```diff
@@ -4476,3 +4476,115 @@ fn test_unicode_string_literal() {
         }
     }
 }
+
+fn check_arrow_precedence(sql: &str, arrow_operator: BinaryOperator) {
+    assert_eq!(
+        pg().verified_stmt(sql),
+        Statement::Query(Box::new(Query {
+            with: None,
+            body: Box::new(SetExpr::Select(Box::new(Select {
+                distinct: None,
+                top: None,
+                projection: vec![SelectItem::UnnamedExpr(Expr::BinaryOp {
+                    left: Box::new(Expr::BinaryOp {
+                        left: Box::new(Expr::Identifier(Ident {
+                            value: "foo".to_string(),
+                            quote_style: None,
+                        })),
+                        op: arrow_operator,
+                        right: Box::new(Expr::Value(Value::SingleQuotedString("bar".to_string()))),
+                    }),
+                    op: BinaryOperator::Eq,
+                    right: Box::new(Expr::Value(Value::SingleQuotedString("spam".to_string()))),
```

Comment:
You could avoid the verbosity by using `verified_expr` instead of `verified_stmt` here

Reply:
It is a great idea -- I made a PR with this change: https://github.com/sqlparser-rs/sqlparser-rs/pull/1367